File: parse_json.kt
import java.io.File

fun main() {
    val htmlFile = File("build/reports/tests/test/classes/webView.actionPlanAdmin.ActionPlanAdminTest.html")
    val text = htmlFile.readText()
    val startIndex = text.indexOf("Full User Data JSON: {")
    if (startIndex != -1) {
        val jsonStart = text.indexOf("{", startIndex)
        val endIndex = text.indexOf("</pre>", jsonStart)
        if (endIndex != -1) {
            val jsonStr = text.substring(jsonStart, endIndex).replace("&quot;", "\"")
            
            // let's just find where `"userProfile"` or `"user_profile"` is
            val idx1 = jsonStr.indexOf("userProfile\"")
            val idx2 = jsonStr.indexOf("user_profile\"")
            if (idx1 != -1) println("Contains userProfile at $idx1")
            if (idx2 != -1) println("Contains user_profile at $idx2")
            
            // extract the first 1000 chars near `"data":{"`
            var dIdx = jsonStr.indexOf("\"data\":{")
            if (dIdx != -1) {
                dIdx = jsonStr.indexOf("\"data\":{", dIdx + 1)
                if (dIdx != -1) {
                    println(jsonStr.substring(dIdx, dIdx + 1000.coerceAtMost(jsonStr.length - dIdx)))
                }
            }
        }
    } else {
        println("Could not find JSON")
    }
}

File: parse_json.py
import re

with open("build/reports/tests/test/classes/webView.actionPlanAdmin.ActionPlanAdminTest.html", "r") as f:
    text = f.read()

start = text.find("Full User Data JSON:")
if start != -1:
    json_start = text.find("{", start)
    end = text.find("</pre>", json_start)
    if end != -1:
        json_str = text[json_start:end].replace("&quot;", "\"")

        # Let's locate the position of "userProfile" in the string
        up_idx = json_str.find('"userProfile"')
        print("Index of userProfile:", up_idx)
        
        # Let's find "data":{"data":
        dd_idx = json_str.find('"data":{"')
        print("Index of data.data:", dd_idx)
        
        # where is the end of data.data?
        # we can't easily parse without failing on JSON errors.
        # But we can just see if up_idx > dd_idx.
        
        # We can also search for the context around it
        if up_idx != -1:
             print("Context before userProfile:", json_str[max(0, up_idx - 100): up_idx])
             print("Context after userProfile:", json_str[up_idx: up_idx + 100])
